### DISCUSSION TITLE
如何通过RPC离线下载文件到网盘目录

### DISCUSSION THREAD

**Discussion by bohong65** [AlistGo/alist#3321] [Q&A]
场景：docker部署alist:latest和aria2-pro:latest，已将alist的data目录映射到aria2-pro中

需求：PC发送RPC请求到aria2-pro，通过指定下载目录为alist中网盘目录的方式，下载到/opt/alist/data/tmp后上传到指定网盘目录并删除tmp中文件，实现通过RPC请求将文件离线下载到网盘

问题：将文件离线下载到网盘是如何实现下载后上传的？我的理解是先下载到/opt/alist/data/tmp目录后上传到远端目录，最后删除tmp目录中的文件。但我并没有源码中找到相关部分（不太熟悉go语言），我应该怎样修改才能实现“通过改变RPC中的下载目录触发【下载后上传并删除】”

有一个不成熟的想法是利用aria2-pro的上传功能，但是aria2-pro的上传应该要通过rclone走webdav的方式，有没有直接上传而不走webdav的方法？

补充：
我知道目前的离线下载到网盘是进入网盘目录后点击离线下载粘贴链接
![image](https://user-images.githubusercontent.com/14939297/217719191-e1d066e6-4939-4f20-a111-fb1cb980cf4c.png)

![image](https://user-images.githubusercontent.com/14939297/217719100-c80aa753-3d8f-4654-b39f-174975e9718a.png)

但希望通过RPC替代粘贴链接，也算是一个新功能请求吧 @xfoxfu

**Comment by xfoxfu** [comment]
See `internal/aria2/monitor.go`. I am not a member of alist, just providing the location based on my own understanding. Correct me if anything wrong.

**Reply by bohong65** [reply]
Oops my mistake, thank you for answering! It's very inspiring and I will work on it.

**Comment by zxy160** [comment]
好奇这个方案有没有实现
